core/src/util/registry.rs

```diff
@@ -0,0 +1,114 @@
+#[cfg(target_os = "windows")]
+pub mod windows {
+    use crate::util::dsn::windows::DSNOpts;
+    use file_dbg_macros::*;
+    use std::io;
+    use winreg::{enums::HKEY_LOCAL_MACHINE, RegKey};
```

Comment:
I am now unsure if we can by-pass calling SQLWriteDSNToIni and SQLWritePrivateProfileString because the root key is different depending on the type of DSN created :'(
It is HKEY_LOCAL_MACHINE for System DSN and HKEY_CURRENT_USER for user DSN. 
SQLGetConfigMode could tell us : https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlgetconfigmode-function
SQLConfigDataSource does not pass the info to ConfigDSN but maps it to User DSN .... https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlconfigdatasource-function?view=sql-server-ver16#comments

Reply:
SQLGetConfigMode is queryable and tells us the mode. We may end up having to go this route, as using SQLWritePrivateProfileString and SQLGetPrivateProfileString are extremely... flaky. This is something easier to show synchronously, but they have unpredictable behavior it seems.